lib/scope.js

```diff
@@ -52,5 +62,64 @@ module.exports = {
     // Plain old waiting. Yeah, I abstracted it into here so only
     // one thing was being called at the end
     if ( options.waitForTimeout ) { await scope.page.waitFor( options.waitForTimeout ); }
-  },
+  },  // Ends afterStep
+
+  getSpecifiedChoice: async function getSpecifiedChoice(scope, {ordinal, label_text, group_label_text, roles}) {
+    /* Returns the elementHandle of the field specified by the arguments
+    * 
+    * @param {string} [ordinal='first'] Which instance of the field on the page.
+    * @param {string} [label_text=''] Text of the label for the field, if it has one.
+    * @param {string} [group_label_text=''] Text of the label for the group of fields to which the field belongs, if there is one.
+    * @param {array} role_types 'radio', 'checkbox', or both. The type of field that the developer could be asking for.
+    * 
+    * @returns instance of ElementHandle or test fails
+    * 
+    * Methodologies proposed:
+    *     * Proposed: First dig all the way down to the label, then dig back out
+    *     * Regected: It's not clear how to keep things in order and filtering them
+    *     that way is more confusing, not less. It also takes more lines of code.
+    *     * Proposed: First collect all elements of expected status.
+    *     * Rejected: That ruins getting the correct ordinal
+    */
+
+    group_label_text = group_label_text || '';
+    // If possible, get the first group to which this choice belongs
+    let group_name = null;
+    if ( group_label_text ) {
+      // Example: `//*[@id="daform"]//label[contains(text(), "serve")][1]`
+      let identifier_xpath = `//*[@id="daform"]//label[contains(text(), "${ group_label_text }")][1]`;
+      let [group_label] = await scope.page.$x( identifier_xpath );
+      expect( group_label.constructor.name, `Did not find a "${ group_label_text }" group on this page. ${ group_label.constructor.name }` ).to.equal('ElementHandle');
+
+      group_name = await group_label.evaluate(( elem ) => elem.getAttribute('for'));
+    }
+
+    let role_type = '';
+    for (let role of roles) {
+      role_type += `@role="${ role }"`;
+      if ( roles.indexOf(role) < roles.length - 1 ) { role_type += ' or '; }
```

Comment:
This might be a personal preference, but I think this line is confusing. I think this might be one of the rare cases where a `for (var i = 0; i < roles.length; i++)` loop is clearer, because then this comparison becomes much simpler.